frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx

```diff
@@ -0,0 +1,130 @@
+import { getStylesRef, rem } from "@mantine/core";
+import type {
+  ButtonStylesParams,
+  ContextStylesParams,
+  MantineTheme,
+  MantineThemeOverride,
+} from "@mantine/core";
+
+export const getButtonOverrides = (): MantineThemeOverride["components"] => ({
+  Button: {
+    defaultProps: {
+      loaderProps: {
+        size: "1rem",
+        color: "currentColor",
+      },
+    },
+    styles: (
+      theme: MantineTheme,
+      { compact }: ButtonStylesParams,
+      { variant }: ContextStylesParams,
+    ) => {
+      const styles = getButtonVariantStyles(theme, variant);
+
+      return {
+        root: {
+          height: "auto",
+          padding: compact ? `${rem(3)} ${rem(7)}` : `${rem(11)} ${rem(15)}`,
+          fontSize: theme.fontSizes.md,
+          lineHeight: "1rem",
+          color: styles.color,
+          borderColor: styles.borderColor,
+
+          "&:disabled": {
+            color: styles.disabledColor,
+            borderColor: styles.disabledBorderColor,
+            backgroundColor: styles.disabledBackgroundColor,
+          },
+          "&[data-loading]": {
+            "&::before": {
+              backgroundColor: styles.loadingBackgroundColor,
+            },
+
+            [`& .${getStylesRef("icon")}`]: {
+              color: styles.loaderColor,
+            },
+          },
+          "&:hover": {
+            color: styles.hoverColor,
+            borderColor: styles.hoverBorderColor,
+            backgroundColor: styles.hoverBackgroundColor,
+          },
+          [`&:has(.${getStylesRef("label")}:empty)`]: {
+            padding: compact ? `${rem(3)} ${rem(3)}` : `${rem(11)} ${rem(11)}`,
+
+            [`.${getStylesRef("icon")}`]: {
+              marginLeft: 0,
+              marginRight: 0,
+            },
+          },
+        },
+        label: {
+          ref: getStylesRef("label"),
+        },
+        icon: {
+          ref: getStylesRef("icon"),
+        },
+        leftIcon: {
+          marginRight: theme.spacing.sm,
+        },
+        rightIcon: {
+          marginLeft: theme.spacing.sm,
+        },
+      };
+    },
+  },
+});
+
+const getButtonVariantStyles = (theme: MantineTheme, variant?: string) => {
+  const hoverColor = theme.fn.rgba(theme.fn.primaryColor(), 0.88);
```

Comment:
This is how currently we compute the hover color for the existing `Button`. Please note that it doesn't match `Hover` color in Figma and this makes buttons look somewhat inconsistent.